server/src/apub/user_inbox.rs

```diff
@@ -25,37 +13,38 @@ pub async fn user_inbox(
   request: HttpRequest,
   input: web::Json<UserAcceptedObjects>,
   path: web::Path<String>,
-  db: web::Data<Pool<ConnectionManager<PgConnection>>>,
+  db: DbPoolParam,
+  _chat_server: ChatServerParam,
 ) -> Result<HttpResponse, Error> {
   // TODO: would be nice if we could do the signature check here, but we cant access the actor property
   let input = input.into_inner();
   let conn = &db.get().unwrap();
-  debug!(
-    "User {} received activity: {:?}",
-    &path.into_inner(),
-    &input
-  );
+  let username = path.into_inner();
+  debug!("User {} received activity: {:?}", &username, &input);
 
   match input {
-    UserAcceptedObjects::Create(c) => handle_create(&c, &request, conn),
-    UserAcceptedObjects::Update(u) => handle_update(&u, &request, conn),
-    UserAcceptedObjects::Accept(a) => handle_accept(&a, &request, conn),
+    UserAcceptedObjects::Create(c) => handle_create(&c, &request, &username, &conn),
+    UserAcceptedObjects::Update(u) => handle_update(&u, &request, &username, &conn),
+    UserAcceptedObjects::Accept(a) => handle_accept(&a, &request, &username, &conn),
   }
 }
 
 /// Handle create activities and insert them in the database.
 fn handle_create(
   create: &Create,
   request: &HttpRequest,
+  _username: &str,
   conn: &PgConnection,
 ) -> Result<HttpResponse, Error> {
-  let community_uri = create
+  // TODO before this even gets named, because we don't know what type of object it is, we need
+  // to parse this out
```

Comment:
Not sure what you are trying to say with this comment.

Reply:
Write now that function is only handling post creation, it needs to be able to handle tons of different types of objects. I think for now we can assume its users doing almost all the actions (except for accepting a follow), but we'll also need to determine the type of actor as well.